src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,181 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
+    private int size = 0;
+
+    public MyLinkedList() {
+        head = new Node<>(null, null, null);
+        tail = new Node<>(null, null, null);
+        head.next = tail;
+        tail.prev = head;
+    }
+
     @Override
     public void add(T value) {
+        if (head.item == null) {
```

Comment:
better compare size
and each time you add a value you should create a new Node, in the beginning head and tail should be null